mantidproject/mantid
# Add 16 new fitfunctions

**Description of work.**
This PR is a spin-out from #26973 - with fixed tests, and the removal of 4 fitting functions (SCgapXXXX) which will be dealt with in a separate issue. 

New fitting functions
    1) AFMLF.py
    2) AFMZF.py
    3) CompositePCRmagnet.py
    4) GauBroadGauKT.py
    5) GaussBessel.py
    6) LowTFMuonium.py
    7) ModOsc.py
    8) MuMinusExpTF.py
    9) PCRmagRedfield.py
    10) PCRmagnetZFKT.py
    11) PCRmagnet.py
    12) PCRmagnetform.py
    13) RedfieldCutoff.py
    14) ZFdipole.py
    15) ZFelectronDipole.py
    16) ZFprotonDipole.py

**To test:**
(1) Test that the fitting function works
(2) Check that the documentation for each fitfunction is correct (e.g formula, variables) 
(3) Confirm that the documentation matches the function described by the source
(4) Test that the function is correctly implemented and that it corresponds with the documentation


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
